### ISSUE TITLE
Providing a tuple for Upsampling scale_factor causes an error even though tuples are supported.

### ISSUE THREAD

**Issue by kureta** [pytorch/pytorch#18308]
## To Reproduce

1. Just call `nn.UpsamplingNearest2d(scale_factor=(2., 1.))`

    TypeError: float() argument must be a string or a number, not 'tuple'

## Expected behavior

return an upsampling layer that only stretches the image on the x axis.

Cause of the problem is in torch.nn.modules.upsampling line 129. It tries to directly convert the `scale_factor` argument into float and fails if it was a tuple.

**Comment by kureta** [comment]
I have created a pull request to resolve this issue. #18310

**Comment by rgommers** [comment]
This was fixed in #20581, for `Upsample` and therefore also for `UpsamplingNearest2d`.

Verifying that:

```
In [1]: import torch                                                                

In [2]: x = torch.arange(1, 5, dtype=torch.float32).view(1, 1, 2, 2)                

In [3]: x                                                                           
Out[3]: 
tensor([[[[1., 2.],
          [3., 4.]]]])

In [4]: m = torch.nn.UpsamplingNearest2d(scale_factor=(2., 1.))                     

In [5]: m(x)                                                                        
/home/rgommers/anaconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/modules/upsampling.py:129: UserWarning: nn.UpsamplingNearest2d is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.{} is deprecated. Use nn.functional.interpolate instead.".format(self.name))
Out[5]: 
tensor([[[[1., 2.],
          [1., 2.],
          [3., 4.],
          [3., 4.]]]])
```

Given that this functionality is already deprecated and the PR  that fixed it didn't add a test, I don't think it helps to do that now. So closing. Thanks @kureta
